Preprocessing/adcpBinMappingPP.m

```diff
@@ -162,9 +163,12 @@
             % let's now interpolate data values at nominal bin height for
             % each profile
             nonMappedData = sample_data{k}.variables{j}.data;
+            
             mappedData = NaN(size(nonMappedData), 'single');
             for i=1:nSamples
-                mappedData(i,:) = interp1(nonMappedHeightAboveSensor(i,:), nonMappedData(i,:), mappedHeightAboveSensor(i,:));
+%                 mappedData(i,:) = interp1(nonMappedHeightAboveSensor(i,:), nonMappedData(i,:), mappedHeightAboveSensor(i,:));
+                %the LeanInterp code is nearly twice as quick as interp1
+                mappedData(i,:) = LeanInterp(nonMappedHeightAboveSensor(i,:), nonMappedData(i,:), mappedHeightAboveSensor(i,:));
```

Comment:
Did you try the griddedInterpolant function? 
```
for i=1:nSamples
    interpFun = griddedInterpolant(nonMappedHeightAboveSensor(i,:),nonMappedData(i,:));
    mappedData(i,:) = interpFun(mappedHeightAboveSensor(i,:));
end
```
I'm getting ~7x faster than interp1 for a double vector with 10k entries interpolated to other ~10k positions. I couldn't test the LeanInterp with typical values, appears less flexible and more memory-hungry!

Another simple optimization is to clip the nonmapped data to the range of mappedHeights, which can reduce even more if we got "out of bounds" data.